instant.ts

```diff
@@ -74,6 +74,38 @@ async function runTests(path: string) {
   }
 }
 
+const orderPackageIds = (allPackages, chosenPackageIds) => {
+  function resolveDeps(id, currentStack) {
+    if (currentStack.includes(id)) throw Error(`Circular dependency present for id ${id}`)
+    currentStack.push(id)
+
+    if (allPackages[id] && allPackages[id].metadata) {
+      if (
+        !allPackages[id].metadata.dependencies ||
+        !allPackages[id].metadata.dependencies.length
+      ) return [id]
+    } else {
+      throw Error(`Package ${id} does not exist or the metadata is invalid`)
+    }
+
+    let orderedIds = []
+
+    allPackages[id].metadata.dependencies.forEach(dependency => {
+      const ids = resolveDeps(dependency, currentStack)
```

Comment:
Just an improvement: I think here we should rather create a copy of the `currentStack` each time the loop runs. This way when the current stack changes it won't affect other loops and we don't have to reset it.